### ISSUE TITLE
[Bug]: Shields Fully Blocking Skill Damage

### ISSUE THREAD

**Issue by lordsrubka5** [promcteam/prorpgitems#114]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The special shield blocking system with block chance and block power works great with physical punch damage however the shields fully block all skill type damage from skillapi and mythicmobs. They do put it on cooldown but the damage is fully ignored.

### Expected Behavior

For shield system to work properly for all damage types

### Environment

```markdown
-
```

**Comment by lordsrubka5** [comment]
Update, it does damage it just doesn't do any hurt animation

**Comment by lordsrubka5** [comment]
Update, rpgitems shielding does not detect skillapi damage as incoming damage and does not trigger the cooldown meaning that it will forever keep blocking. Did not test with non 100% block powers

**Comment by Travja** [comment]
I was attempting to test this out with only one account but am currently unable to do so xD. It looks like I can't even set up a skill to damage myself unless it's using True Damage, which doesn't do any knockback or damage animation anyway. So I'm going to have to find some other way to test this out some more to track down a solution.

**Comment by Travja** [comment]
Update: It looks like the way we set up as a result of #95, skills actually don't go through ProRPGItem's damage/dodge/etc functionality at all. Unfortunately, enabling this functionality re-introduced problems we already resolved. I may be able to address this in a way that we can get the benefit of ProRPGItem's blocking/dodging mechanics without causing issues, but this will be potentially sort of difficult to tackle.
